SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc

```diff
@@ -0,0 +1,967 @@
+    if (detid == Dead_detID){
+      isbad = true;
+      break;
+    }
+  }
+  
+  if (!isbad) return;
+
+  signal_map_type& theSignal = _signal[detID]; // check validity
+  
+    
+  for (auto & s : theSignal) {
+    std::pair<int,int> ip;
+    if (pixelFlag) ip  = PixelDigi::channelToPixel(s.first);
+    else ip  = Phase2TrackerDigi::channelToPixel(s.first);//get pixel pos
+
+    if (Module == "whole") s.second.set(0.); // reset amplitude
+    else if (Module == "tbmA" && ip.first >= 80 && ip.first <= 159) s.second.set(0.);
+    else if (Module == "tbmB" && ip.first <= 79) s.second.set(0.);
+  }
+}
+// ==========================================================================
+void Phase2TrackerDigitizerAlgorithm::module_killing_DB(uint32_t detID) {
+  bool isbad = false;
+  
+  std::vector<SiPixelQuality::disabledModuleType>disabledModules = SiPixelBadModule_->getBadComponentList();
+  
+  SiPixelQuality::disabledModuleType badmodule;
+  for (size_t id = 0;id < disabledModules.size(); id++) {
+    if (detID == disabledModules[id].DetID) {
+      isbad = true;
+      badmodule = disabledModules[id];
+      break;
+    }
+  }
+  
+  if (!isbad)
+    return;
+
+  signal_map_type& theSignal = _signal[detID]; // check validity
+  
+  if (badmodule.errorType == 0) { // this is a whole dead module.
+    for (auto & s : theSignal) {
+      s.second.set(0.); // reset amplitude
+    }
+  }
+  else { // all other module types: half-modules and single ROCs.
+    // Get Bad ROC position:
+    // follow the example of getBadRocPositions in CondFormats/SiPixelObjects/src/SiPixelQuality.cc
+    std::vector<GlobalPixel> badrocpositions (0);
+    for (unsigned int j = 0; j < 16; j++) {
+      if (SiPixelBadModule_->IsRocBad(detID, j) == true) {
+	std::vector<CablingPathToDetUnit> path = map_.product()->pathToDetUnit(detID);
+	for (auto const & p : path) {
+          const PixelROC* myroc = map_.product()->findItem(p);
+          if (myroc->idInDetUnit() == j) {
+	    LocalPixel::RocRowCol  local = {39, 25};   //corresponding to center of ROC row, col
+	    GlobalPixel global = myroc->toGlobal(LocalPixel(local));
+	    badrocpositions.push_back(global);
+	    break;
+	  }
+	}
+      }
+    }
+    
+    for (auto & s : theSignal) {
+      std::pair<int,int> ip;
+      if (pixelFlag) ip  = PixelDigi::channelToPixel(s.first);
+      else ip = Phase2TrackerDigi::channelToPixel(s.first);
+      
+      for (auto const & p : badrocpositions) {
+	if (p.row >= 80 && ip.first >= 80) {
+	  if ((fabs(ip.second - p.col) < 26) ) {s.second.set(0.);}
+          else if (p.row == 120 && ip.second-p.col == 26) {s.second.set(0.);}
+          else if (p.row == 119 && p.col-ip.second == 26) {s.second.set(0.);}
+	}
+	else if (p.row < 80 && ip.first < 80 ) {
+	  if ((fabs(ip.second - p.col) < 26) ) {s.second.set(0.);}
+          else if(p.row == 40 && ip.second-p.col == 26) {s.second.set(0.);}
+          else if(p.row == 39 && p.col-ip.second == 26) {s.second.set(0.);}
```

Comment:
pls identify 80, 120, 26, 40, 39, 119 as constants so they are easy to find and change when the time comes.